src/sage/combinat/partition.py

```diff
@@ -7497,8 +7513,24 @@ def first(self):
 
             sage: Partitions(3, starting=[2,1]).first()
             [2, 1]
+            sage: Partitions(3, starting=[1,1,1]).first()
+            [1, 1, 1]
+            sage: Partitions(3, starting=[1,1]).first()
+            False
+            sage: Partitions(3, starting=[3,1]).first()
+            [3]
+            sage: Partitions(3, starting=[2,2]).first()
+            [2, 1]
         """
-        return self._starting
+        if self._starting in Partitions_n(self.n):
```

Comment:
There is a much lighter way to test this:
```suggestion
        if sum(self._starting) == n:
```

Reply:
Thanks